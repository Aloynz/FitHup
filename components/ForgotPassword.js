import React, { useState } from "react";
import {
  StatusBar,
  StyleSheet,
  Text,
  TextInput,
  TouchableOpacity,
  View,
} from "react-native";
import { SafeAreaView } from "react-native-safe-area-context";
import colours from "../assets/colours/colours";
import { useFonts } from "expo-font";
import Style from "./Style";
import MailIcon from "./MailIcon";

export default ForgotPassword = ({ navigation }) => {
  const [loaded] = useFonts({
    RobotoMedium: require("../assets/fonts/Roboto-Medium.ttf"),
    RobotoRegular: require("../assets/fonts/Roboto-Regular.ttf"),
    MontserratBold: require("../assets/fonts/Montserrat-Bold.ttf"),
  });

  {
    /*Email*/
  }
  const [email, setEmail] = useState("");

  return (
    <View style={Style.container}>
      <StatusBar />
      {/* Header */}
      <SafeAreaView style={Style.header}>
        <Text style={Style.headerText}>Forgot{"\n"}Password?</Text>
      </SafeAreaView>

      {/*Body */}
      <View style={Style.body}>
        {/*Email Field */}
        <View>
          <Text style={Style.email}>Email address</Text>

          {/* Email Rectangle */}
          <View style={Style.rect}>
            {/* Mail icon */}
            <MailIcon/>
            <TextInput
              style={Style.sampleEmail}
              placeholder="Enter your email"
              placeholderTextColor={colours.text}
              onChangeText={(email) => setEmail(email)}
              autoComplete="email"
            />
          </View>

          <Text style={styles.resetPassword}>
            *We will send a link to your email to reset your password.
          </Text>
        </View>

        {/*Continue button*/}

        <View style={[Style.loginOrSignUpButton, { marginTop: 130 }]}>
          <Button title ={"Continue"}/>
        </View>

        {/*Back to Login */}
        <View>
          <Text style={[Style.account, { marginLeft: 70 }]}>
            Remember your password?
          </Text>

          <TouchableOpacity onPress={() => navigation.navigate("Login")}>
            <Text style={[Style.signUpOrLogin, { marginLeft: 250 }]}>
              Log in
            </Text>
          </TouchableOpacity>
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  resetPassword: {
    color: colours.text,
    fontFamily: "RobotoRegular",
    fontSize: 11,
    marginLeft: 27,
    marginTop: 12,
  },
});
