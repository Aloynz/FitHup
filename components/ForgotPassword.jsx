import React, { useState } from "react";
import {
  Alert,
  StatusBar,
  StyleSheet,
  Text,
  TextInput,
  TouchableOpacity,
  View,
} from "react-native";
import { supabase } from '../lib/supabase'
import { SafeAreaView } from "react-native-safe-area-context";
import colours from "../assets/colours/colours";
import { useFonts } from "expo-font";
import Style from "./Style";
import MailIcon from "./MailIcon";
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';
import Button from './Button';

export default ForgotPassword = ({ navigation }) => {
  const [loaded] = useFonts({
    RobotoMedium: require("../assets/fonts/Roboto-Medium.ttf"),
    RobotoRegular: require("../assets/fonts/Roboto-Regular.ttf"),
    MontserratBold: require("../assets/fonts/Montserrat-Bold.ttf"),
  });

  {
    /*Email*/
  }
  const [email, setEmail] = useState("");
  const [loading, setLoading] = useState(false)

  async function sendResetEmail() {
    try {
      setLoading(true)

      const { data, error } = await supabase.auth.api
        .resetPasswordForEmail(email)

      if (error) {
        throw error
      } else if (data) {
        Alert.alert("Password reset email sent successfully");
      }
    } catch (error) {
      Alert.alert(error.message);
    } finally {
      setLoading(false);
    }
  }

  return (
    <View style={Style.container}>
      <StatusBar />
      {/* Header */}
      <SafeAreaView style={Style.header}>
        <Text adjustsFontSizeToFit style={Style.headerText}>Forgot{"\n"}Password?</Text>
      </SafeAreaView>

      {/*Body */}
      <KeyboardAwareScrollView
        style={Style.body}
        resetScrollToCoords={{ x: 0, y: 0 }}
        scrollEnabled={true}
        contentContainerStyle={{ flexGrow: 1 }} >
        {/*Email Field */}
        <View>
          <Text style={Style.email}>Email address</Text>

          {/* Email Rectangle */}
          <View style={Style.rect}>
            {/* Mail icon */}
            <MailIcon />
            <TextInput
              testID="forgotPwdEmail"
              style={Style.sampleEmail}
              placeholder="Enter your email"
              placeholderTextColor={colours.text}
              value={email}
              onChangeText={(email) => setEmail(email)}
              autoComplete="email"
              autoCapitalize="none"
            />
          </View>

          <Text style={styles.resetPassword}>
            *We will send a link to your email to reset your password.
          </Text>
        </View>

        <View style={[Style.loginOrSignUpButtonContainer]}>
          {/*Continue button*/}
          <View style={[Style.loginOrSignUpButton]}>
            <Button
              testID="forgotPwdBtn"
              title={"Continue"}
              disabled={loading}
              onPress={() => sendResetEmail()}
            />
          </View>

          {/*Back to Login */}
          <View style={Style.signUpOrLoginContainer}>
            <Text style={[Style.account]}>
              Remember your password?
            </Text>

            <TouchableOpacity onPress={() => navigation.navigate("Login")}>
              <Text testID="loginBtn" style={[Style.signUpOrLogin]}>
                Log in
              </Text>
            </TouchableOpacity>
          </View>
        </View>
      </KeyboardAwareScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  resetPassword: {
    color: colours.text,
    fontFamily: "RobotoRegular",
    fontSize: 11,
    marginLeft: 27,
    marginTop: 12,
  },
});
