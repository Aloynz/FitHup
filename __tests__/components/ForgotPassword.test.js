import React from 'react';
import ForgotPassword from '../../components/ForgotPassword'
import renderer from 'react-test-renderer';
import { render, waitFor, act } from '@testing-library/react-native';

describe('Forgot Password', () => {
  const forgotPassword = renderer.create(
    <ForgotPassword />
  ).toJSON;

  it('should render successfully', async () => {
    await act(async () => {
      expect(forgotPassword).toBeDefined();
    })
  });

  it('should render correctly', async () => {
    await act(async () => {
      expect(forgotPassword).toMatchSnapshot();
    })
  });

  const { getByTestId } = render(<ForgotPassword />);

  it('should have email field', () => {
    waitFor(() => {
      const forgotPwdEmail = getByTestId('forgotPwdEmail');
      expect(forgotPwdEmail).toBeInTheDocument();
    })
  });

  it('should have continue button', () => {
    waitFor(() => {
      const forgotPwdBtn = getByTestId('forgotPwdBtn');
      expect(forgotPwdBtn).toBeInTheDocument();
    })
  });

  it('should have back to login button', () => {
    waitFor(() => {
      const loginBtn = getByTestId('loginBtn');
      expect(loginBtn).toBeInTheDocument();
    })
  });
});