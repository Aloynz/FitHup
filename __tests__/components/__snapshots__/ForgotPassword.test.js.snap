// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forgot Password should render correctly 1`] = `[Function]`;
